src/ucs/type/worker_param.h

```diff
@@ -0,0 +1,19 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 20016.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_TYPE_CPU_PARAM_H
```

Comment:
at first I tried to do it in ucp. I ran into a lot of depenency issues.
hence I asked @shssf for help. He gave me a sample which I used. this is the sample.